rts/idris_utf8.h

```diff
@@ -1,24 +1,38 @@
 #ifndef _IDRIS_UTF8
 #define _IDRIS_UTF8
 
+#include <stdint.h>
+#include <stdlib.h>
+
 /* Various functions for dealing with UTF8 encoding. These are probably
    not very efficient (and I'm (EB) making no guarantees about their
    correctness.) Nevertheless, they mean that we can treat Strings as
    UFT8. Patches welcome :). */
 
+typedef uint8_t utf8_byte;
+
+// Pascal-style strings with their lengths.
+typedef struct {
+    size_t size;
+    utf8_byte* str_data;
+} SizedString;
```

Comment:
The cost of this is that the code gets more complex. Given that the `str_data` array is always allocated immediately adjacent to the `SizedString`, it should be in the same cache line. Would we really win much there to make it worth the more convoluted code?